cmds/core/kexec/kexec_linux.go

```diff
@@ -92,11 +95,33 @@ func main() {
 			log.Fatal(err)
 		}
 		defer mbkernel.Close()
+		var r io.ReaderAt
+		// kernel files are sometimes gzip'ed, and there's no good
+		// pattern to the naming. Just try to read it as a gzip,
+		// and if it fails, proceed with the original.
+		if f, err := gzip.NewReader(mbkernel); err == nil {
+			defer f.Close()
+			// We need a ReaderAt, and it's best to find out
+			// right away if the gunzip will not work out.
+			b, err := ioutil.ReadAll(f)
+			if err != nil {
+				log.Fatalf("Reading gzip'ed kernel: %v", err)
```

Comment:
To not confuse people when looking at the logs, I'd prefer to provide a message more in the direction of
```go
  log.Fatalf("Could not read as gzipped kernel, will try to read in plain. Error was: %v", err)
```

Reply:
This is the case where the file was known to be gzip'ed, but the read failed for some reason. When you get here, the file is known to be a gzip'ed file, but the read of it has failed. It may be damaged but we do know it had the signature of a gzip'ed file. There's no point in trying to read it as plain.

Reply:
hence the log.Fatalf: this is an exit.

Reply:
Right, misread that :)